hydroflow/src/scheduled/input.rs

```diff
@@ -1,31 +1,54 @@
-use std::{cell::RefCell, rc::Rc};
+use std::{cell::RefCell, marker::PhantomData, rc::Rc, sync::mpsc::SyncSender};
 
 use super::{OpId, Reactor};
 
-pub struct Input<T> {
+pub trait Givable<T> {
+    fn give(&self, t: T) -> bool;
+}
+
+impl<T> Givable<T> for Rc<RefCell<Vec<T>>> {
```

Comment:
Don't love the `Input<T, Rc<RefCell<Vec<T>>>>`  compared to `Input<T, SyncSender<T>>` but is ok for now. Maybe mark TODO(mignwei) if you agree?

Reply:
Gave it a slightly friendlier name.